File: 0235-lowest-common-ancestor-of-a-binary-search-tree/0235-lowest-common-ancestor-of-a-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */

class Solution {
    public TreeNode lowestCommonAncestor(TreeNode r, TreeNode p, TreeNode q) {
        int min=Math.min(p.val,q.val);
        int max=Math.max(p.val,q.val);
        while(r!=null){
            int c=r.val;
            if(min <r.val && max<r.val){
                r=r.left;
            }
            else if(min >r.val && max>r.val){
                r=r.right;
            }
            else{
                break;
            }
        }
        return r;
    }
}

File: 0235-lowest-common-ancestor-of-a-binary-search-tree/0235-lowest-common-ancestor-of-a-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def lowestCommonAncestor(self, r: 'TreeNode', p: 'TreeNode', q: 'TreeNode') -> 'TreeNode':
        mi=min(p.val,q.val)
        ma=max(p.val,q.val)
        while r!=None:
            if(mi<r.val and ma<r.val):
                r=r.left
            elif(mi>r.val and ma>r.val):
                r=r.right
            else:
                break
        return r